src/ui/BlendEngine.js

```diff
@@ -142,8 +146,19 @@ exports = Class(View, function () {
     this._canvas = new Canvas({ width: MAX_TEX_WIDTH, height: MAX_TEX_HEIGHT, useWebGL: true });
   };
 
-  this.obtainParticleArray = function (count) {
-    for (var i = 0; i < count; i++) {
+  this.obtainParticleArray = function (count, opts) {
+    var isBrowser = (userAgent.APP_RUNTIME === 'browser');
+    var isMobile = (userAgent.DEVICE_TYPE === 'mobile');
+    var isSimulator = userAgent.SIMULATED;
+   
+    opts = opts || {};
+
+    // TODO: disable blend engine on mobile browsers until the performance is improved
+    var currCount = (isBrowser && isMobile && !isSimulator)
```

Comment:
Slight adjustment here. We only want to disable `BlendEngine` if `performanceScore` is not provided in the `opts`. Otherwise, they should continue to function normally - we don't want it to suddenly stop working if someone is using it.